src/librustdoc/passes/calculate_doc_coverage.rs

```diff
@@ -245,11 +252,16 @@ impl fold::DocFolder for CoverageCalculator {
                 );
 
                 let has_doc_example = tests.found_tests != 0;
+                let hir_id = self.ctx.tcx.hir().local_def_id_to_hir_id(i.def_id.expect_local());
+                let (level, source) = self.ctx.tcx.lint_level_at_node(MISSING_DOCS, hir_id);
+                let should_have_docs =
+                    level != lint::Level::Allow || !matches!(source, LintSource::Node(..));
```

Comment:
It looks like this overrides `-D missing_docs`. Is there a reason you did that?

Reply:
It doesn't override anything as far as I can tell: if the lint level is `Allow`, then it doesn't require to have docs on it, but if it's the "default" (which is the case currently), meaning you didn't yourself add `#[allow(...)]`, then it needs to have documentation.

Reply:
There are three variants of `LintSource`, not two: https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/lint/enum.LintSource.html#variant.CommandLine. If you want to exclude `Default`, you should say that explicitly:
```suggestion
                    level != lint::Level::Allow || matches!(source, LintSource::Default);
```

Reply:
Oh I see, great catch!